spydrnet_physical/util/tests/test_ConnectPointList.py

```diff
@@ -0,0 +1,219 @@
+from email.mime import base
+import pstats
+import unittest
+from spydrnet_physical.util import ConnectPoint , ConnectPointList
+
+class test_connectpointlist(unittest.TestCase):
+
+	def setUp(self):
+		self.pts = ConnectPointList(5,10)
+
+
+	def test_cursor_point(self):
+		'''This test checks whether the cursor is correctly placed initially at the right coordinates using cursor_point'''
+		self.pts.cursor = (1,1)
+		self.assertEqual(self.pts.cursor,(1,1))
+
+	def test_getx(self):
+		'''This test checks whether getx correctly provides the x coordinate of the cursor'''
+		self.pts.cursor = (3,4)
+		self.assertEqual(self.pts.get_x,3)
+
+	def test_gety(self):
+		'''This test checks whether gety correctly provides the y coordinate of the cursor'''
+		self.pts.cursor = (3,4)
+		self.assertEqual(self.pts.get_y,4)
+
+	def test_set_cursor(self):
+		'''This test checks whether the set_cursor correcly places cursor at the new coordinates'''
+		self.pts.cursor = (1,1)
+		self.pts.set_cursor(5,5)
+		self.assertEqual(self.pts.cursor,(5,5))
+
+	def test_set_color(self):
+		'''This test checks whether the color of connect point list is set correcly'''
+		pt = ConnectPoint(1, 2, 1, 3, level="up")
+		pts = ConnectPointList(5,10,pt)
+		pts.set_color("red")
+		self.assertEqual(pt.color,"red")
+
+	def test_search_from_point(self):
+		'''This test checks whether the search from point correctly returns connection going out of the given coordinate'''
+		self.pts.cursor = (2,3)
+		self.pts.move_x(1)
+		self.pts.move_y(1)
+		point = self.pts.search_from_point((3,3))
+		self.assertEqual(point.connection,(3,3,3,4))
+
+	def test_search_to_point(self):
+		'''This test checks whether the search to point correctly returns connection coming into the given coordinate'''
+		self.pts.cursor = (2,3)
+		self.pts.move_x(1)
+		self.pts.move_y(1)
+		point = self.pts.search_to_point((3,3))
+		self.assertEqual(point.connection,(2,3,3,3))
+
+	def test_push_connection_down(self):
+		'''This test checks whether the level of the connection is going one down'''
+		pt = ConnectPoint(2, 2, 2, 3)
+		self.pts.push_connection_down (pt)
+		self.assertEqual(pt.level,"down")
+
+	def test_pull_connection_up(self):
+		'''This test checks whether the level of the connection is going one down'''
+		pt = ConnectPoint(2, 2, 2, 3)
+		self.pts.pull_connection_up(pt)
+		self.assertEqual(pt.level,"up")
+
+	def test_make_top_connection(self):
+		'''This test checks whether the connection is made with the top level'''
+		pass
+		#pts = ConnectPointList(5,10)
+		#pt = ConnectPoint(2, 2, 2, 3)
+		#pts.make_top_connection()
+		#self.assertEqual(pt.level,"same")
+
+	def test_reset_level(self):
+		'''This test checks whether the connection reset to the same level'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		self.pts.reset_level(pt)
+		self.assertEqual(pt.level,"same")
+
+	def test_hold_cursor(self):
```

Comment:
Also, check what is the default (after initialization) value of this state

Reply:
Do you mean that we check if the cursor is in hold or release state when initialized ?

Reply:
If by default/ initialization you mean that what is the cursor value before cursor is defined, then the cursor value is [ ].